### PR TITLE
Oc 323 profile view

### PR DESCRIPTION
**Pull request by qclin**
draft profile view 

- cleaned up the existing divs for the biography 
- reused ViewCollection with userId props

### CODE REVIEW

File: src/actions/user/viewProfile.tsx [hunk 1]
```diff
@@ -1,51 +1,60 @@
-import { API } from 'aws-amplify';
+import { API } from "aws-amplify";
 import {
   FETCH_PROFILE,
   FETCH_PROFILE_ERROR,
   FETCH_PROFILE_ERROR_NO_SUCH_PROFILE
-} from '../../reducers/user/viewProfile';
-import { LOADINGOVERLAY } from '../loadingOverlay';
+} from "../../reducers/user/viewProfile";
```
Comment (Dan-Wood) [inline src/actions/user/viewProfile.tsx]: Just note, the linter should've complained, or maybe it's Webstorm (my local IDE) but double quotes outside of JSX it has a fit about it :(

File: src/components/user/profile/ViewProfile.tsx [hunk 2]
```diff
@@ -33,13 +36,13 @@ class ViewProfile extends React.Component<Props, State> {
...
-      this.setState({ errorMessage: 'No profile with that id.' });
+      this.setState({ errorMessage: "No profile with that id." });
     }
   }
 
   render() {
-    if (typeof this.props.profile === 'undefined') {
-      return 'Loading...';
+    if (typeof this.props.profile === "undefined") {
+      return "Loading...";
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: We should probably show an error message here, more than likely the user doesn't exist.

File: src/components/user/profile/ViewProfile.tsx [hunk 3]
```diff
@@ -50,64 +53,95 @@ class ViewProfile extends React.Component<Props, State> {
...
-              <Col xs="12" className="details border">
-                Website: {website}
-              </Col>
-              <Col xs="12" className="details border">
-                Position: {position}
-              </Col>
-              <Col xs="12" className="details border">
-                Affiliation: {affiliation}
+                <div className="align-bottom profession">
+                  <div> {field_expertise} </div>
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: Some of these fields may not exist such as field_expertise and website to name a few.

File: src/components/user/profile/ViewProfile.tsx [hunk 4]
```diff
@@ -50,64 +53,95 @@ class ViewProfile extends React.Component<Props, State> {
...
+              <div>
+                Location:
+                <span className="ml-auto">
+                  {city}, {country}
+                </span>
+              </div>
+              <div>
+                Website:
+                <span className="ml-auto">
+                  <a
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: If this field doesn't exist (it'll return null from the API) we'll still have a clickable link, we may as well just hide it all :)

File: src/components/user/profile/ViewProfile.tsx [hunk 5]
```diff
@@ -50,64 +53,95 @@ class ViewProfile extends React.Component<Props, State> {
...
+                    href={website || ""}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                  >
+                    {website}
+                  </a>
+                </span>
+              </div>
+              <div>
+                Position: <span className="ml-auto">{position}</span>
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: This also may not exist.

Have a look here, https://github.com/AcrossTheCloud/TBA21-API/blob/master/schema/schema.sql#L265

  profile_image varchar(1024) -- could be null
  featured_image varchar(1024) -- could be null
  field_expertise varchar(256) -- could be null
  city varchar(128) -- could be null
  country varchar(128) -- could be null
  biography varchar(1024) -- could be null
  website varchar(2048) -- could be null
  social_media varchar(256)[], -- could be null
  public_profile boolean, false or true (we don't show false profiles to the public)
  affiliation varchar(256), -- could be null
  position varchar(256), -- could be null

// These fields are only applicable if the profile_type is one of 'Collective', 'Institution'
  contact_person varchar (256), -- could be null
  contact_position varchar (256), -- could be null
  contact_email varchar (256), -- could be null

File: src/components/user/profile/ViewProfile.tsx [hunk 6]
```diff
@@ -50,64 +53,95 @@ class ViewProfile extends React.Component<Props, State> {
...
+                  {tags.map(ea => (
+                    <span className="tags">{ea}</span>
+                  ))}
+                </p>
+              </div>
+            </div>
           </Col>
         </Row>
+        <h4> Contributed Items </h4>
+        <ViewCollection userId="4"/>
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: Nice reuse! :)

File: src/components/user/profile/ViewProfile.tsx [hunk 7]
```diff
@@ -50,64 +53,95 @@ class ViewProfile extends React.Component<Props, State> {
...
       </div>
     );
   }
 }
 
 // State to props
-const mapStateToProps = (state: { viewProfile: State }) => { // tslint:disable-line: no-any
+const mapStateToProps = (state: { viewProfile: State }) => {
+  // tslint:disable-line: no-any
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: Left overs.

File: src/styles/layout/elements.scss [hunk 8]
```diff
@@ -0,0 +1,15 @@
+span.tags {
```
Comment (Dan-Wood) [inline src/styles/layout/elements.scss]: Just make sure this doesn't conflict with the homepage and collection / item view pages.

File: src/styles/fonts/typography.scss [hunk 9]
```diff
@@ -0,0 +1,9 @@
+@import "../base/variables";
+@import "fonts.scss";
+
+.caption-text {
+  font-family: "Roboto Mono", monospace;
```
Comment (Dan-Wood) [inline src/styles/fonts/typography.scss]: Overall font is Roboto Mono on the body so no need for this here.

File: src/styles/components/pages/viewProfile.scss [hunk 10]
```diff
@@ -11,28 +11,29 @@
...
+    a {
+      color: $subline-gray;
+      :hover {
+        color: $soft-gray;
       }
     }
+  }
 
+  .biography-section {
+    display: inline-grid;
```
Comment (Dan-Wood) [inline src/styles/components/pages/viewProfile.scss]: Partial support for IE11

https://caniuse.com/#search=inline-grid

File: src/router.tsx [hunk 11]
```diff
@@ -144,15 +144,15 @@ export const AppRouter = () => {
...
-              {/*<Route*/}
-              {/*  path="/profiles/:profileId"*/}
-              {/*  render={() => (*/}
-              {/*    <div className="container-fluid main blue">*/}
-              {/*      <ViewProfile />*/}
-              {/*      <Footer />*/}
-              {/*    </div>)*/}
-              {/*  }*/}
-              {/*/>*/}
+              <Route
```
Comment (Dan-Wood) [inline src/router.tsx]: Nice catch! I should have mentioned to branch off Dev 👍

File: src/components/user/profile/ViewProfile.tsx [hunk 12]
```diff
@@ -1,30 +1,33 @@
...
   fetchProfile: Function;
   profile: Profile;
 }
 
 class ViewProfile extends React.Component<Props, State> {
-  matchedId: string = '';
+  matchedId: string = "";
 
-  constructor(props: any) { // tslint:disable-line: no-any
+  constructor(props: any) {
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: Should be Props no any, this was from a previous commit.

File: src/components/user/profile/ViewProfile.tsx [hunk 13]
```diff
@@ -1,30 +1,33 @@
...
+
+import "styles/components/pages/viewProfile.scss";
+import { Alerts, ErrorMessage } from "../../utils/alerts";
+import { Profile } from "../../../types/Profile";
+import { fetchProfile } from "../../../actions/user/viewProfile";
+import { Col, Row } from "reactstrap";
 
 interface Props extends RouteComponentProps, Alerts {
   fetchProfile: Function;
   profile: Profile;
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: This becomes optional if you change the bad any code below.

**Comment by Dan-Wood** [comment]
Thanks Qiao just a few nit picks :) most of it was left overs from the WIP I believe.

File: src/components/user/profile/ViewProfile.tsx [hunk 14]
```diff
@@ -50,64 +53,95 @@ class ViewProfile extends React.Component<Props, State> {
...
+                Position: <span className="ml-auto">{position}</span>
+              </div>
+              <div>
+                Affiliation: <span className="ml-auto">{affiliation}</span>
+              </div>
+              <div>
+                Most used concept tags
+                <p className="mt-5">
+                  {tags.map(ea => (
+                    <span className="tags">{ea}</span>
```
Comment (Dan-Wood) [inline src/components/user/profile/ViewProfile.tsx]: Missing key

https://reactjs.org/docs/lists-and-keys.html

Can use the index from map()

File: src/styles/fonts/typography.scss (hunk 9 above)
Comment (qclin) [reply src/styles/fonts/typography.scss]: i'll like to built a typography file for all the use of types so we can insert easily later

File: src/components/user/profile/ViewProfile.tsx (hunk 5 above)
Comment (qclin) [reply src/components/user/profile/ViewProfile.tsx]: good to know

File: src/actions/user/viewProfile.tsx (hunk 1 above)
Comment (qclin) [reply src/actions/user/viewProfile.tsx]: ah i'll adjust my ide autoformatter for single quotes on this repo

File: src/styles/components/pages/viewProfile.scss (hunk 10 above)
Comment (qclin) [reply src/styles/components/pages/viewProfile.scss]: it'll just fallback to the row flex

File: src/router.tsx (hunk 11 above)
Comment (qclin) [reply src/router.tsx]: ah okay, i can rebase DEV if needed

File: src/components/user/profile/ViewProfile.tsx (hunk 2 above)
Comment (qclin) [reply src/components/user/profile/ViewProfile.tsx]: this should be handled in the api call, resolve reject . not in the view

File: src/styles/layout/elements.scss (hunk 8 above)
Comment (qclin) [reply src/styles/layout/elements.scss]: the home page is a div, but we should create generic components for reuse

Comment (Dan-Wood) [reply src/styles/layout/elements.scss]: Agreed.

File: src/styles/components/pages/viewProfile.scss (hunk 10 above)
Comment (Dan-Wood) [reply src/styles/components/pages/viewProfile.scss]: 👍

File: src/components/user/profile/ViewProfile.tsx (hunk 2 above)
Comment (Dan-Wood) [reply src/components/user/profile/ViewProfile.tsx]: 👍

**Comment by qclin** [comment]
Okay! will try that, I simply reused the CollectionSlider, Since the mockup had fixed unit gray boxes. Can explore a flex layout to adapt to content ratio.

**Comment by qclin** [comment]
In order to reuse the Home page scroll view, I had to refactor a bit. what happened in the last 3 commits. 

Extracted view components from `actions/home.tsx` into stand-alone components
    - Layout > feed/FeedItem
    - HighLightsLayout > highlight/HighlightItem, HighlightPreview, AnnouncementView
    - HighlightsItemDetails > highlight/HighlightItemDetails
    - HomePageAudioPreview > Feed/FeedAudioPreview
move Home type into types/
move DateFromTimeYearProduced into it’s own utils componen


Clicking around works, I found that I might have broken the AudioPreview on click `FeedAudioPreview L39`

@Dan-Wood  please have a look first as it's quite a lot of shifting and would be great if you can catch anything that might've been broken. Since we don't have unit tests yet

**Comment by matthewberryman** [comment]
If you could do a tiny bit more work @qclin to resolve the merge conflicts then Dan can review Monday.

**Comment by qclin** [comment]
@Dan-Wood can you review this PR, it has some conflict of file placement with the other open PR  #152 would like to merge it soon, it's been two weeks. Thanks

**Comment by Dan-Wood** [comment]
@qclin I've merged Announcements, dev and staging into this branch.

However, a few things that your changes has introduced.

1. Highlights aren't showing
2. Audio waveforms and PDF's are flashing on re-render
3. Infinite loop causing browser crashes **(fixed)**
4. Reshuffle on re-render (occasional)
5. No profile exists, shows loading text
6. No Contributed Items shows text
7. No Most used concept tags still shows box
8. No image, still has col with empty space

File: src/actions/collections/viewCollection.tsx [hunk 15]
```diff
@@ -18,49 +18,6 @@ export const FETCH_COLLECTION_ERROR_NO_SUCH_COLLECTION =
...
-  });
-
-  try {
-    // TODO: this returns an array of collections
-    const { collections } = await queryByUuid(uuid);
-
-    if (!!collections && collections.length) {
-      Promise.all(
-
-        collections.map(async collection => {
```
Comment (qclin) [inline src/actions/collections/viewCollection.tsx]: @Dan-Wood  i just saw that you removed all changes to fetch A profile collections, 
it's that what the API endpoint returns ? and that's what this call is also doing ?

Comment (Dan-Wood) [reply src/actions/collections/viewCollection.tsx]: Yep the collections, I haven't finished on this yet, you're getting the items in the collection which is incorrect.

The scope was to show the items and collections belonging to that profile, not all the items in each collection. As items don't have to live in a collection. Pretty much wanted the same as the homepage where it's a masonry view  of the items and collections.

Comment (qclin) [reply src/actions/collections/viewCollection.tsx]: the endpoint returns collections that the profile was part of, which contains items that are not authored by the user, so what exactly is a contributed Item, that the Profile has co-authorship? shouldn't we fetch from items then? 
/items?coauthor=uuid&author=uuid

**Comment by matthewberryman** [comment]
stale PR

**Comment by Dan-Wood** [comment]
🍞

**Pull request closed without merging**